xbmc/platform/win32/WIN32Util.cpp

```diff
@@ -1424,6 +1424,99 @@ HDR_STATUS CWIN32Util::GetWindowsHDRStatus()
   return status;
 }
 
+/*!
+ * \brief Retrieve from the system the max luminance of SDR content in HDR.
+ *
+ * Retrieve from the system the max luminance of SDR content in HDR.
+ * Note: always returns 80 nits when the screen is in SDR mode.
+ *
+ * \param gdiDeviceName The screen to retrieve the information for
+ * \param sdrWhiteLevel Max luminance in nits, clamped to 10000
+ * \return true if a value could be read from the system and copied to sdrWhiteLevel, false otherwise
+*/
+bool CWIN32Util::GetSystemSdrWhiteLevel(const std::wstring& gdiDeviceName, unsigned int* sdrWhiteLevel)
+{
+#ifdef TARGET_WINDOWS_STORE
+  auto displayInformation = DisplayInformation::GetForCurrentView();
+
+  if (displayInformation)
+  {
+    auto advancedColorInfo = displayInformation.GetAdvancedColorInfo();
+
+    if (advancedColorInfo)
+    {
+      auto sdrnits = advancedColorInfo.SdrWhiteLevelInNits();
+      if (sdrWhiteLevel)
+        *sdrWhiteLevel = static_cast<unsigned int>(sdrnits);
```

Comment:
I would like use `float` for sdrWhiteLevel in whole chain as in origin (Windows API) is float:

https://learn.microsoft.com/en-us/uwp/api/windows.graphics.display.advancedcolorinfo.sdrwhitelevelinnits?view=winrt-22621

and at end is also float (shader):

https://github.com/xbmc/xbmc/blob/e87cf69d373cc133f1ca632aa404e744af777b11/xbmc/guilib/GUIShaderDX.h#L109

Only exception is for manual GUI setting: It was expressly chosen int to have 100 steps and use % numeric control. Then use only static_cast to convert manual setting and all the rest in float.

Reply:
I had it that way earlier, will revert.